lib/mmonit/connection.rb

```diff
@@ -40,7 +40,11 @@ def connect
 		end
 
 		def login
-			self.request('/z_security_check', "z_username=#{@username}&z_password=#{@password}", true).code.to_i == 302
+			self.request('/z_security_check', URI.encode_www_form({'z_username'=>@username, 'z_password'=>@password, 'z_csrf_protection'=>'off'}), true).code.to_i == 302
+		end
+
+		def logout
```

Comment:
I'm sorry, I didn't test the logout method properly. It seems to be that the 'logout' request returns an incorrectly compressed response. This is fixed by requesting an uncompressed response, by setting the 'Accept-Encoding: identity' header. (similar bug: https://github.com/jaimeiniesta/metainspector/issues/125)